### PR TITLE
Adding "named_port" support to Google's instance_group_manager.

### PR DESCRIPTION
**Pull request by chrusty**
This allows HTTP and HTTPs load-balancers to direct traffic to ports other than tcp/80 and tcp/443.

### CODE REVIEW

File: builtin/providers/google/resource_compute_instance_group_manager.go [hunk 1]
```diff
@@ -165,6 +207,22 @@ func resourceComputeInstanceGroupManagerRead(d *schema.ResourceData, meta interf
 	d.Set("target_size", manager.TargetSize)
 	d.Set("self_link", manager.SelfLink)
 
+	// Prepare the list of named ports
+	namedPortsCount := d.Get("named_port.#").(int)
+
+	namedPorts := make([]map[string]interface{}, 0, 1)
```
Comment (billf) [inline builtin/providers/google/resource_compute_instance_group_manager.go]: if these are returned in a different order than the schema stored them in, will `reflect.DeepEqual()` as used in `helper/schema/resource_data.go` see them as equal?

e.g. 
`{0: {"name": "http", "port": 8080}, 1: {"name": "https", "port" 8443}}`
`{0: {"name": "https", "port": 8443}, 1: {"name": "http", "port" 8080}}`

Comment (chrusty) [inline builtin/providers/google/resource_compute_instance_group_manager.go]: I've not encountered issues with it yet, and I've used this branch to build instance_group_managers on 2 different GCE projects (in some tests with up to 10 named-ports). The order which seems to matter is within the .tf files themselves. I've repeatedly run "terraform apply" after the initial run and never seen anything being re-ordered.

Is this what you're asking?

**Comment by chrusty** [comment]
I've used this to build several managed instance-groups in a number of different Google Compute projects, and it seems to work as intended. I was able to bring up managed-instance-groups with named_ports, and bring up "backends" behind HTTP / HTTPs load-balancers which use those ports by name.

**Comment by jen20** [comment]
Hi @chrusty! I've squashed the commits and merged thes in 9aa8bbda. Thanks for your contribution!

**Pull request closed without merging**